packages/trader/src/Modules/Trading/Containers/trade-chart.tsx

```diff
@@ -0,0 +1,200 @@
+import { observer, useStore } from '@deriv/stores';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { ChartBottomWidgets } from './chart-widgets';
+import SmartChartSwitcher from './smart-chart-switcher';
+import AccumulatorsChartElements from '../../SmartChart/Components/Markers/accumulators-chart-elements';
+import ToolbarWidgets from '../../SmartChart/Components/toolbar-widgets';
+import ToolbarWidgetsBeta from '../../SmartChartBeta/Components/toolbar-widgets';
+import AllMarkers from '../../SmartChart/Components/all-markers.jsx';
+import type { TBottomWidgetsParams } from './trade';
+
+type TTradeChartProps = {
+    bottomWidgets?: (props: TBottomWidgetsParams) => React.ReactElement;
+    is_accumulator: boolean;
+    topWidgets: () => React.ReactElement;
+};
+
+const TradeChart = observer((props: TTradeChartProps) => {
+    const { is_accumulator, topWidgets } = props;
+    const { client, ui, common, contract_trade, portfolio } = useStore();
+    const {
+        accumulator_barriers_data,
+        accumulator_contract_barriers_data,
+        chart_type,
+        granularity,
+        has_crossed_accu_barriers,
+        markers_array,
+        updateChartType,
+        updateGranularity,
+    } = contract_trade;
+    const { all_positions } = portfolio;
+    const { is_chart_countdown_visible, is_chart_layout_default, is_dark_mode_on, is_mobile, is_positions_drawer_on } =
+        ui;
+    const { current_language, is_socket_opened } = common;
+    const { currency, is_beta_chart, should_show_eu_content } = client;
+    const {
+        active_symbols,
+        barriers_flattened: extra_barriers,
+        chartStateChange,
+        chart_layout,
+        exportLayout,
+        has_alternative_source,
+        is_trade_enabled,
+        main_barrier_flattened: main_barrier,
+        setChartStatus,
+        show_digits_stats,
+        symbol,
+        wsForget,
+        wsForgetStream,
+        wsSendRequest,
+        wsSubscribe,
+    } = useTraderStore();
+
+    const settings = {
+        countdown: is_chart_countdown_visible,
+        isHighestLowestMarkerEnabled: false, // TODO: Pending UI,
+        language: current_language.toLowerCase(),
+        position: is_chart_layout_default ? 'bottom' : 'left',
+        theme: is_dark_mode_on ? 'dark' : 'light',
+        ...(is_accumulator ? { whitespace: 190, minimumLeftBars: is_mobile ? 3 : undefined } : {}),
+    };
+
+    const { current_spot, current_spot_time } = accumulator_barriers_data || {};
+
+    const getBottomWidgets = React.useCallback(
+        ({ digits, tick }: TBottomWidgetsParams) => (
+            <ChartBottomWidgets digits={digits} tick={tick} show_accumulators_stats={is_accumulator} />
+        ),
+        [is_accumulator]
+    );
+
+    const getMarketsOrder = (active_symbols: ActiveSymbols): string[] => {
+        const synthetic_index = 'synthetic_index';
+
+        const has_synthetic_index = !!active_symbols.find(s => s.market === synthetic_index);
```

Comment:
If a boolean needed, that `.some` can be used instead of `.find`
Please, ignore if not relevant

Reply:
@kate-deriv it's applicable👍🏻